src/routes/components/button2/+page.svelte

```diff
@@ -0,0 +1,223 @@
+		example).
+	</Typography>
+{/snippet}
+
+{#snippet examples()}
+	<p class="example-row">
+		<Typography variant="small" bold>1. Strong button</Typography>
+		<Button variant="strong">Strong button</Button>
+		<Button variant="strong"><Checkmark size={24} />Confirm</Button>
+		<Button variant="strong">Proceed<ArrowRight size={24} /></Button>
+	</p>
+
+	<p class="example-row">
+		<Typography variant="small" bold>2. Outline button</Typography>
+		<Button variant="secondary">Button</Button>
+		<Button variant="secondary"><Close size={24} />Cancel</Button>
+		<Button variant="secondary">Next<ArrowRight size={24} /></Button>
+	</p>
+
+	<p class="example-row">
+		<Typography variant="small" bold>3. Ghost button</Typography>
+		<Button variant="ghost">Ghost button</Button>
+		<Button variant="ghost"><Information size={24} />Info</Button>
+		<Button variant="ghost">More<ChevronRight size={24} /></Button>
+	</p>
+
+	<p class="example-row">
+		<Typography variant="small" bold>4. Strong icon button</Typography>
+		<Button variant="strong"><Save size={24} /></Button>
+	</p>
+
+	<p class="example-row">
+		<Typography variant="small" bold>5. Icon button</Typography>
+		<Button variant="ghost"><OverflowMenuVertical size={24} /></Button>
+	</p>
+
+	<p class="example-row">
+		<Typography variant="small" bold>6. Dark overlay button</Typography>
+		<Button variant="darkoverlay">Overlay</Button>
+		<Button variant="darkoverlay"><Checkmark size={24} />Done</Button>
+		<Button variant="darkoverlay">Next<ArrowRight size={24} /></Button>
+	</p>
+
+	<p class="example-row">
+		<Typography variant="small" bold>7. Light overlay button</Typography>
+		<Button variant="overlay">Overlay</Button>
+		<Button variant="overlay"><Launch size={24} />Open</Button>
+		<Button variant="overlay">See all<CaretRight size={24} /></Button>
+	</p>
+
+	<p class="example-row">
+		<Typography variant="small" bold>8. Dark overlay icon button</Typography>
+		<Button variant="darkoverlay"><TrashCan size={24} /></Button>
+	</p>
+
+	<p class="example-row">
+		<Typography variant="small" bold>8. Light overlay icon button</Typography>
+		<Button variant="overlay"><SidePanelOpen size={24} /></Button>
+	</p>
+{/snippet}
+
+{#snippet controls()}
+	<Select bind:value={variant} placeholder="Button type">
+		<Option value="strong">Strong button</Option>
+		<Option value="secondary">Outline button</Option>
+		<Option value="ghost">Ghost button</Option>
+	</Select>
+
+	<Select bind:value={dimension} placeholder="Button size">
+		<Option value="default">Default</Option>
+		<Option value="large">Large</Option>
+		<Option value="compact">Compact</Option>
+		<Option value="small">Small</Option>
+	</Select>
+
+	<Switch
+		checked={leftIcon}
+		label="Left icon"
+		onchange={(e) => leftIcon = (e?.target as HTMLInputElement).checked}
+	></Switch>
+	<Switch
+		checked={rightIcon}
+		label="Right icon"
+		onchange={(e) => rightIcon = (e?.target as HTMLInputElement).checked}
```

Comment:
This typecasting is quite weird, why do we need it? Is there a problem with the `Switch` component? I guess `Switch` does not have the `checked` bindable, and tat is why you need to use `onchange`. Still the typecasting feels wrong.

Reply:
Indeed, I used `onchange` because the `checked` is not bindable. I fixed it by making it bindable in `Switch`. The typecasting on `onchange` could be fixed by using `currentTarget` instead of target like this:

```svelte
		onchange={(e) => leftIcon = e.currentTarget.checked}
```